core/src/main/java/org/n52/iceland/binding/kvp/KvpBinding.java

```diff
@@ -180,7 +189,12 @@ public String getRequestParameterValue(Map<String, String> map) throws OwsExcept
         Decoder<AbstractServiceRequest<?>, Map<String, String>> decoder = getDecoder(k);
         LOGGER.trace("Using {} to decode paramers: {}", decoder, Arrays.toString(parameterValueMap.entrySet().toArray()));
         if (decoder != null) {
-            return decoder.decode(parameterValueMap);
+            AbstractServiceRequest<?> request = decoder.decode(parameterValueMap);
+            if (includeOriginal) {
+                request.setOriginalRequest(Joiner.on(Constants.QUERSTIONMARK_STRING)
```

Comment:
Please remove `QUESTIONMARK_STRING` and just use `"?"`. Also the `Joiner` should be a `static final` member.